src/views/virtualmachines/tree/components/TreeViewRightClickActionMenu/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { NavigateFunction } from 'react-router-dom-v5-compat';
+
+import { ActionDropdownItemType } from '@kubevirt-utils/components/ActionsDropdown/constants';
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getLabel, getNamespace } from '@kubevirt-utils/resources/shared';
+import { FOLDER_SELECTOR_PREFIX, VM_FOLDER_LABEL } from '@virtualmachines/tree/utils/constants';
+import { vmsSignal } from '@virtualmachines/tree/utils/signals';
+
+export const getCreateVMAction = (
+  navigate: NavigateFunction,
+  namespace: string,
+): ActionDropdownItemType => ({
+  cta: () => navigate(`/k8s/ns/${namespace}/catalog`),
+  id: 'create-vm',
+  label: t('Create VirtualMachine'),
+});
+
+export const getVMsTrigger = (triggerElement: HTMLElement | null) => {
+  if (!triggerElement) return [];
+
+  const [prefix, namespace, folderName] = triggerElement?.id?.split('/') || ['', '', ''];
```

Comment:
`triggerElement?.id?.split('/')` appears to be used a few times. Consider making it a function with a default value argument:

`const [prefix, namespace, folderName] = getTriggerElementComponents(triggerElement, ['', '', '']);`